File: CloudflareFileStorage.go
package main

import (
    "fmt"
    "io/ioutil"
    "mime/multipart"
    "net/http"
    "os"
    "path/filepath"
)

type CloudflareFileStorage struct {
    WorkerURL string
}

func (c *CloudflareFileStorage) UploadFile(filePath string) (string, error) {
    file, err := os.Open(filePath)
    if err != nil {
        return "", err
    }
    defer file.Close()

    body := &multipart.Writer{}
    req, err := http.NewRequest("POST", fmt.Sprintf("%s/upload", c.WorkerURL), body)
    req.Header.Set("Content-Type", body.FormDataContentType())
    req.MultipartForm = map[string][]string{"file": {file.Name()}}

    client := &http.Client{}
    res, err := client.Do(req)
    if err != nil {
        return "", err
    }
    defer res.Body.Close()

    result, err := ioutil.ReadAll(res.Body)
    return string(result), err
}

func (c *CloudflareFileStorage) DownloadFile(fileName, savePath string) ([]byte, error) {
    res, err := http.Get(fmt.Sprintf("%s/download/%s", c.WorkerURL, fileName))
    if err != nil {
        return nil, err
    }
    defer res.Body.Close()

    body, err := ioutil.ReadAll(res.Body)
    if err != nil {
        return nil, err
    }

    saveFile := filepath.Join(savePath, fileName)
    err = ioutil.WriteFile(saveFile, body, 0644)
    return body, err
}

func (c *CloudflareFileStorage) DeleteFile(fileName string) (string, error) {
    req, err := http.NewRequest("DELETE", fmt.Sprintf("%s/delete/%s", c.WorkerURL, fileName), nil)
    if err != nil {
        return "", err
    }

    client := &http.Client{}
    res, err := client.Do(req)
    if err != nil {
        return "", err
    }
    defer res.Body.Close()

    result, err := ioutil.ReadAll(res.Body)
    return string(result), err
}

File: CloudflareFileStorage.rb
require 'net/http'
require 'uri'
require 'json'
require 'fileutils'

class CloudflareFileStorage
  def initialize(worker_url)
    @worker_url = worker_url.chomp('/')
  end

  def upload_file(file_path)
    raise "File not found: #{file_path}" unless File.exist?(file_path)

    uri = URI("#{@worker_url}/upload")
    file = File.open(file_path, 'rb')
    req = Net::HTTP::Post.new(uri)
    req.set_form([['file', File.open(file_path)]], 'multipart/form-data')

    res = Net::HTTP.start(uri.hostname, uri.port) { |http| http.request(req) }
    file.close
    res.body
  end

  def download_file(file_name, save_path)
    uri = URI("#{@worker_url}/download/#{file_name}")
    res = Net::HTTP.get_response(uri)
    raise "Error downloading file: #{res.message}" unless res.is_a?(Net::HTTPSuccess)

    file_path = File.join(save_path, file_name)
    File.write(file_path, res.body)
    res.body
  end

  def delete_file(file_name)
    uri = URI("#{@worker_url}/delete/#{file_name}")
    req = Net::HTTP::Delete.new(uri)
    res = Net::HTTP.start(uri.hostname, uri.port) { |http| http.request(req) }
    res.body
  end
end
